cmd/controller-manager/app/controller-manager.go

```diff
@@ -212,16 +215,91 @@ func getFederationConfig(opts *options.Options) *corev1a1.FederationConfig {
 		glog.Fatalf("Cannot decode FederationConfig from file %q: %v", federationConfig, err)
 	}
 
+	// set to current namespace to make sure `FederationConfig` is updated in correct namespace
+	fedConfig.Namespace = opts.Config.FederationNamespace
 	glog.Infof("Setting Options with FederationConfig from file %q: %v", federationConfig, fedConfig.Spec)
 	return fedConfig
 }
 
+func getDuration(specValue metav1.Duration, defaultValue time.Duration) time.Duration {
+	if specValue.Duration == 0 {
+		return defaultValue
+	}
+	return specValue.Duration
+}
+
+func getString(specValue, defaultValue string) string {
+	if specValue == "" {
+		return defaultValue
+	}
+	return specValue
+}
+
+func getInt(specValue, defaultValue int) int {
+	if specValue == 0 {
+		return defaultValue
+	}
+	return specValue
+}
+
+func setDefaultFederationConfig(fedConfig *corev1a1.FederationConfig) {
```

Comment:
Setting defaults via the federation config likely removes the need to set defaults [here](https://github.com/kubernetes-sigs/federation-v2/blob/master/cmd/controller-manager/app/options/options.go#L40).

Reply:
Yes. It does not need anymore.

Reply:
yes. Addressed.